deployments/pheno/services/apiv1/src/endpoints/trainings.py

```diff
@@ -0,0 +1,815 @@
+                    with zipfile.ZipFile(io.BytesIO(file.file.read()), "r") as zip_ref:
+                        zip_ref.extractall(tmp_dir)
+                else:
+                    out_path = os.path.join(tmp_dir, file.filename)
+                    async with async_open(out_path, "wb") as out_file:
+                        content = await file.read()  # async read
+                        await out_file.write(content)  # async write
+
+            # list files, and upload, protects against any other file structure in the zipfile
+            for root, _, files in os.walk(tmp_dir):
+                for filename in files:
+                    file_path = os.path.join(root, filename)
+                    # if valid image
+                    if "__MACOSX" not in file_path and await is_image_file(file_path):
+                        if ".csv" in filename.lower():
+                            try:
+                                pathFileOut = file_path.replace("csv", "png").replace("CSV", "png")
+                                signals = readCSV(file_path)
+                                makePNG(signals, pathFileOut)
+                                file_path = pathFileOut
+                            except:
+                                print("error csv to png")
+
+                        relFileNoExt, file_ext = os.path.splitext(filename)
+                        unique_name = relFileNoExt.replace("/", "_") + "_" + str(uuid.uuid4())[:8] + file_ext
+                        if filename == "sample.jpg":  # upload sample image
+                            sample_img_key = f"trainings/{user_id}/{study_name}/{name}/sample.jpg"
+                            response = upload_file_to_s3(sample_img_key, file_path)
+                        else:  # upload class images
+                            key = (
+                                f"trainings/{user_id}/{study_name}/{name}/{name}/{class_name}/{unique_name}"
+                                if val_data_source == "combined"
+                                else f"trainings/{user_id}/{study_name}/{name}/{name}/{val_or_train}/{class_name}/{unique_name}"
+                            )
+
+                            response = upload_file_to_s3(key, file_path)
+
+            logger.info(f"Uploading {file}: {response}")
+        return JSONResponse(
+            status_code=status.HTTP_200_OK, content={"message": f"Successfully uploaded images for {name}"}
+        )
+    except Exception as e:
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail=f"Error uploading training images to s3: {e}",
+        )
+
+
+@router.post(
+    "/submitNewTraining/{user_id}",
+    response_description="redirects to step 2/3 and passes certain params to next view",
+    description="Step 1/3 when starting a new training. Route creates new training entry in db with user-inputted data then uploads to s3.",
+)
+async def new_train_submit(
```

Comment:
haven't been able to figure out how to use form data with a Pydantic model to clean this up. I can just send this in the request body if I need to and store it as an object in the FE instead.